spec/requests/api/v1/journeys_controller_index_spec.rb

```diff
@@ -0,0 +1,103 @@
+
+    let(:locations) { create_list(:location, 2, suppliers: [supplier]) }
+    let!(:move) { create(:move, from_location: locations.first, to_location: locations.last) }
+    let!(:last_journey) { create(:journey, move: move, supplier: supplier, client_timestamp: '2020-05-04T09:00:00Z') }
+    let!(:intermediate_journeys) { create_list(:journey, intermediate_journeys_count, move: move, supplier: supplier, client_timestamp: '2020-05-04T08:30:00Z') }
+    let!(:first_journey) { create(:journey, move: move, supplier: supplier, client_timestamp: '2020-05-04T08:00:00Z') }
+    let!(:other_move_journey) { create(:journey, supplier: supplier) } # another journey for a different move, same supplier
+    let!(:other_supplier_journey) { create(:journey, move: move) } # another journey for a different supplier, same move
+
+    let(:intermediate_journeys_count) { 1 }
+    let(:page) { 1 }
+    let(:per_page) { 20 }
+
+    before do
+      get "/api/v1/moves/#{move.id}/journeys?page=#{page}&per_page=#{per_page}", headers: headers, as: :json
+    end
+
+    context 'when successful' do
+      let(:schema) { load_yaml_schema('get_journeys_responses.yaml') }
+
+      it_behaves_like 'an endpoint that responds with success 200'
+
+      describe 'response_json' do
+        subject(:journey_ids) { response_json['data'].map { |x| x['id'] } }
+
+        it { expect(journey_ids.size).to be 3 }
+        it { expect(journey_ids.first).to eql(first_journey.id) } # NB: first_journey should come before last_journey
+        it { expect(journey_ids.last).to eql(last_journey.id) }
+        it { is_expected.not_to include(other_move_journey.id) } # NB: should not contain another move's journey
+        it { is_expected.not_to include(other_supplier_journey.id) } # NB: should not contain another supplier's journey in the same move
+      end
+
+      describe 'paginating results' do
+        let(:intermediate_journeys_count) { 19 }
+
+        describe 'page size' do
+          subject { response_json['data'].size }
+
+          context 'when page=1' do
+            let(:page) { 1 }
+
+            it { is_expected.to be(20) }
+          end
+
+          context 'when page=2' do
+            let(:page) { 2 }
+
+            it { is_expected.to be(1) }
+          end
+
+          context 'when per_page=15' do
+            let(:per_page) { 15 }
+
+            it { is_expected.to be(15) }
+          end
+        end
+
+        it 'provides meta data with pagination' do
+          expect(response_json['meta']['pagination']).to include_json(
+            per_page: 20,
+            total_pages: 2,
+            total_objects: 21,
+            links: {
+              first: "/api/v1/moves/#{move.id}/journeys?page=1&per_page=#{per_page}",
+              last: "/api/v1/moves/#{move.id}/journeys?page=2&per_page=#{per_page}",
+              next: "/api/v1/moves/#{move.id}/journeys?page=2&per_page=#{per_page}",
+            },
+          )
+        end
```

Comment:
Appreciate this would be extra work, but there are a few request specs now that have the same pattern of pagination tests - perhaps worth adding a shared context for these, e.g. `it_behaves_like 'a paginated endpoint'`. See `allocations_controller_index_spec.rb`, `moves_controller_index_spec.rb`, `locations_controller_spec.rb`.

Reply:
This can be done in a future PR

Reply:
yep, I agree it would be good to tidy that up - will do as a separate PR